framework/src/modules/interoperability/mainchain/cc_commands/registration.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { codec } from '@liskhq/lisk-codec';
+import { BaseCCCommand } from '../../base_cross_chain_command';
+import { CROSS_CHAIN_COMMAND_ID_REGISTRATION } from '../../constants';
+import { MainchainInteroperabilityStore } from '../store';
+import { registrationCCMParamsSchema } from '../../schema';
+import { CCCommandExecuteContext, MessageFeeTokenID } from '../../types';
+import { BaseInteroperableAPI } from '../../base_interoperable_api';
+import { createCCMsgBeforeSendContext } from '../../context';
+import { SubStore } from '../../../../node/state_machine/types';
+
+interface CCMRegistrationParams {
+	networkID: Buffer;
+	name: string;
+	messageFeeTokenID: MessageFeeTokenID;
+}
+
+export class MainchainCCRegistrationCommand extends BaseCCCommand {
+	public ID = CROSS_CHAIN_COMMAND_ID_REGISTRATION;
+	public name = 'registration';
+	public schema = registrationCCMParamsSchema;
+	private readonly _interoperableCCAPIs: Map<number, BaseInteroperableAPI>;
+
+	public constructor(moduleID: number, interoperableCCAPIs: Map<number, BaseInteroperableAPI>) {
+		super(moduleID);
+		this._interoperableCCAPIs = interoperableCCAPIs;
```

Comment:
Is this `interoperableCCAPIs` different for mainchain & sidechain command ?
If yes then maybe a distinct variable name might be better to understand.

Reply:
because CCCommands created in interoperable modules will never use or call `_interoperableCCAPIs `, they will only use API of interoperability module to interact

Reply:
Then maybe distinct names for main/side chain would be better to understand. eg `mainchainInteroperableCCAPIs`